File: redis-manager-ui/src/api/board.ts
import RequestHttp from '../utils/request'

namespace Rboard {

    // 登录成功后返回的token
    export interface ResData {
        data: {
            aliredis: number;
            codis: number;
            txredis: number;
            cluster: number;
        }
        errorCode: number;
        msg: string;
    }
  }
  // 用户登录
  export const BoardDesc = () => {
      // 返回的数据格式可以和服务端约定
      return RequestHttp.get<Rboard.ResData>('/board/v1/desc');
  }
  

File: src/rhttp/v1/board.go
package v1

import (
	"net/http"

	"github.com/gin-gonic/gin"
	"github.com/iguidao/redis-manager/src/hsc"
	"github.com/iguidao/redis-manager/src/middleware/mysql"
)

func BoardDesc(c *gin.Context) {
	code := hsc.SUCCESS
	result := make(map[string]int64)
	result["aliredis"] = mysql.DB.GetCloudNumber("aliredis")
	result["txredis"] = mysql.DB.GetCloudNumber("txredis")
	result["codis"] = mysql.DB.GetCodisNumber()
	result["cluster"] = mysql.DB.GetClusterNumber()
	c.JSON(http.StatusOK, gin.H{
		"errorCode": code,
		"msg":       hsc.GetMsg(code),
		"data":      result,
	})
}
